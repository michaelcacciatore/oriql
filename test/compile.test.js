const compile = require('../compile');

const schema = require('./schema/schema');

describe('Compiler', () => {
  it('should compile correctly', async () => {
    const compiled = await compile([schema]);

    expect(compiled).toMatchSnapshot();
  });
});
