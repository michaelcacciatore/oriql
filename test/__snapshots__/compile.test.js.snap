// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiler should compile correctly 1`] = `
Object {
  "client": Object {
    "instances": Object {
      "testQuery": "testQuery",
    },
    "mutations": Object {
      "testQuery": Object {
        "getFirstName": "getFirstName(id: $id) { id employee_name employee_salary employee_age profile_image }",
      },
    },
    "queries": Object {
      "testQuery": "testQuery(foo: $foo, test: $test, id: $id) { foo lol bar test { michael brielle robert(foo: $foo, test: $test, nested: $nested) { itworks theresa } } mom { michael brielle robert(foo: $foo, test: $test, nested: $nested) { itworks theresa } } nestedNoName { hasObject double { isDouble } } firstName(id: $id) employees { id employee_name employee_salary employee_age profile_image } }",
    },
  },
  "flow": undefined,
  "server": GraphQLSchema {
    "__allowedLegacyNames": Array [],
    "__validationErrors": undefined,
    "_directives": Array [
      "@include",
      "@skip",
      "@deprecated",
    ],
    "_implementations": Object {},
    "_mutationType": "RootMutationType",
    "_possibleTypeMap": Object {},
    "_queryType": "RootQueryType",
    "_subscriptionType": undefined,
    "_typeMap": Object {
      "Boolean": "Boolean",
      "MyTestType": "MyTestType",
      "RootMutationType": "RootMutationType",
      "RootQueryType": "RootQueryType",
      "String": "String",
      "__Directive": "__Directive",
      "__DirectiveLocation": "__DirectiveLocation",
      "__EnumValue": "__EnumValue",
      "__Field": "__Field",
      "__InputValue": "__InputValue",
      "__Schema": "__Schema",
      "__Type": "__Type",
      "__TypeKind": "__TypeKind",
      "dad": "dad",
      "dadArgs": "dadArgs",
      "testQuery": "testQuery",
      "testQuery_employees": "testQuery_employees",
      "testQuery_nestedNoName": "testQuery_nestedNoName",
      "testQuery_nestedNoName_double": "testQuery_nestedNoName_double",
    },
    "astNode": undefined,
    "extensionASTNodes": undefined,
  },
  "typescript": undefined,
}
`;
